clar2wasm/src/words/options.rs

```diff
@@ -58,31 +65,17 @@ impl Word for IsNone {
         _expr: &SymbolicExpression,
         args: &[SymbolicExpression],
     ) -> Result<(), GeneratorError> {
-        let opt = args.get_expr(0)?;
-        generator.traverse_expr(builder, opt)?;
-        // there is an optional type on top of the stack.
-
-        // Get the type of the optional expression
-        let ty = generator
-            .get_expr_type(opt)
-            .expect("input expression must be typed")
-            .clone();
-
-        let some_ty = if let TypeSignature::OptionalType(some_type) = &ty {
-            &**some_type
-        } else {
-            panic!("Expected an Optional type. Found: {:?}", ty);
+        match traverse_optional(generator, builder, args) {
+            Ok(_) => {
+                // Add one to stack
+                // and proceed with a XOR operation
+                // to invert the indicator value
+                builder.i32_const(1);
+                builder.binop(BinaryOp::I32Xor);
+            }
+            Err(e) => return Err(e),
         };
```

Comment:
```suggestion
        traverse_optional(generator, builder, args)?;

        // Add one to stack
        // and proceed with a XOR operation
        // to invert the indicator value
        builder.i32_const(1).binop(BinaryOp::I32Xor);
```

Reply:
That would raise a warning. But we can mute it with a silent binding. What you think?

Reply:
Sorry, I missed the question mark early returning an error, if need. 👍